File: movie-ticket-system/payment-service/src/main/java/com/movieticket/payment_service/service/PaymentService.java
package com.movieticket.payment_service.service;

import com.movieticket.payment_service.model.Payment;
import com.movieticket.payment_service.repository.PaymentRepository;

import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class PaymentService {

    private final PaymentRepository repository;

    public PaymentService(PaymentRepository repository) {
        this.repository = repository;
    }

    public Payment createPayment(Payment payment) {

        payment.setStatus("SUCCESS");

        return repository.save(payment);
    }

    public List<Payment> getAllPayments() {
        return repository.findAll();
    }

    public Payment getPayment(String id) {
        return repository.findById(id).orElse(null);
    }

    public List<Payment> getPaymentByBooking(String bookingId) {
        return repository.findByBookingId(bookingId);
    }

    public Payment updateStatus(String id, String status) {

        Payment payment = repository.findById(id).orElse(null);

        if (payment != null) {
            payment.setStatus(status);
            repository.save(payment);
        }

        return payment;
    }

    public void deletePayment(String id) {
        repository.deleteById(id);
    }

}

File: movie-ticket-system/payment-service/src/services/payment.service.js
const Payment = require("../models/payment.model");

const createPayment = async (payload, userId) => {
  const data = { ...payload };
  // Spring behavior: always set SUCCESS on create
  data.status = "SUCCESS";
  if (userId && !data.userId) {
    data.userId = userId;
  }
  const payment = await Payment.create(data);
  return payment;
};

const getAllPayments = async () => {
  return Payment.find().lean();
};

const getPaymentById = async (id) => {
  return Payment.findById(id).lean();
};

const getPaymentsByBookingId = async (bookingId) => {
  return Payment.find({ bookingId }).lean();
};

const updateStatus = async (id, status) => {
  return Payment.findByIdAndUpdate(id, { status }, { new: true }).lean();
};

const deletePayment = async (id) => {
  await Payment.findByIdAndDelete(id);
};

module.exports = {
  createPayment,
  getAllPayments,
  getPaymentById,
  getPaymentsByBookingId,
  updateStatus,
  deletePayment,
};

